Feature/Import Indicators tab from upc branch

nshandra opened the pull request:
## Feature: Import Indicators tab from upc branch
### References

- No specific issue for this task, parent issue: [Incorporate information about individual analysis in WebApp HMIS Dictionary](https://app.clickup.com/t/3771yxt)

### Description
- Imports the Indicators tab to the app, the programs tab changes are ignored except the link code.
- The link functionality add a new dependency: [angular-sessionstorage](https://classic.yarnpkg.com/en/package/angular-sessionstorage)

nshandra committed:
Merge relevant code from 'upc' into 'feature/add_visualizations_to_indicators'. Minify angular-sessionstorage.js into angular-sessionstorage.min.js.

nshandra committed:
Remove merge mistake

nshandra committed:
Switch angular-sessionstorage dependency to yarn

nshandra committed:
Merge branch 'feature/add_visualizations_to_indicators' into feature/add_upc_indicator_tab

adrianq committed:
Merge branch 'development' into feature/add_upc_indicator_tab

adrianq commented on app/datasets/datasets.controllers.js:
```diff
@@ -55,6 +65,33 @@ datasetsModule.controller("datasetsMainController", [
+                endLoadingState(true);
+            });
+        }
+        /*else
```
Can this comment be deleted?

nshandra replied:
Done.

adrianq commented on app/dossierIndicators/dossierIndicators.services.js:
```diff
@@ -0,0 +1,160 @@
+//Récupération de tous les indicateurs : phase d'initialisation
```
Please translate or remove this comment

nshandra replied:
Done. Removed as it wasn't very informative.

adrianq commented on app/dossierIndicators/dossierIndicators.services.js:
```diff
@@ -0,0 +1,160 @@
+]);
+
+var qryIndicatorCategoryOptionCombos =
+    dhisUrl + "categoryOptionCombos.json?filter=id\\:eq\\::id&fields=name&paging=false"; //Modification de displayName par Name
```
Please remove this comment as well

nshandra replied:
Removed.

adrianq commented on yarn.lock:
```diff
@@ -69,6 +69,11 @@
   version "4.5.12"
   resolved "https://codeload.github.com/tinymce/tinymce-dist/tar.gz/3d007282ded510e32c638b2ed5a9ffe1f219d329"
 
+angular-sessionstorage@^1.1.5:
```
Do we actually need this new dependency?

nshandra replied:
Right now its used for the "link to" functionality in the indicators dossier. I'll look into a possible alternative.

adrianq commented on app/dossierPrograms/dossierPrograms.services.js:
```diff
@@ -473,3 +473,35 @@ searchModule.factory("dossiersProgramVisualizationTableFactory", [
         };
     },
 ]);
+
```
If this code is not relevant please remove it

nshandra replied:
Done.

adrianq reviewed (changes requested):
@nshandra I am aware this code is a merge from a different fork but I don't want to merge it as is. Can you please have a look at the comments below? I am particularly interested in understanding if we can avoid adding a new dependency. I will ask @tokland to review it later on

nshandra committed:
Add xlsx temporal dependency

nshandra committed:
Add export to excel button and dossiersProgramDataService service

nshandra committed:
Add export controller

nshandra committed:
Remove leftover code

nshandra committed:
Merge branch 'feature/add_upc_indicator_tab' into feature/add_xlsx_export_to_programs

nshandra committed:
Merge branch 'feature/add_upc_indicator_tab' into feature/add_xlsx_export_to_programs

nshandra committed:
Fix build-bower-deps.sh execution perms

nshandra committed:
Finish program dossier translations tags

nshandra committed:
Add dossiersProgramDataService object structural template

nshandra committed:
Add translations to existing worksheets, add undefined value check

nshandra committed:
Add program stages

adrianq committed:
bump version and update authors

adrianq committed:
fix calculation mode

nshandra committed:
Change stage tabs name format

nshandra committed:
Fix ng-switch-when multi-option not working (incorrect angularjs version)

nshandra committed:
Add makeRulesActionsWorksheet

nshandra committed:
Add translations

nshandra committed:
Switch from "Name" to "Program Rule Name" in makeRulesActionsWorksheet

adrianq committed:
remove strange charact from tab name

nshandra committed:
Delete leftover comments and code.

nshandra committed:
Merge branch 'feature/add_upc_indicator_tab' into feature/add_xlsx_export_to_programs

nshandra committed:
Add eyeseetea/xlsx-populate lib

nshandra committed:
Remove SheetJS lib

nshandra committed:
Switch dossiersProgramExport controller to xlsx-populate lib:
- Add makeStageSectionSheetName function with better logic.
- Add utility functions writeSheetHeader and writeSheetData.

nshandra committed:
Add downloadExcel function tentative

nshandra committed:
Make programIndicators boundaries field less JSONy, prettify

nshandra committed:
Remove testing alert and redundant await

adrianq committed:
Merge pull request #6 from EyeSeeTea/feature/add_xlsx_export_to_programs

Feature/Capacity to download the previous information in Excel
